### ISSUE TITLE
HMR not working with PostHTML Modules

### ISSUE THREAD

**Issue by w-biggs** [parcel-bundler/parcel#3218]
# 🐛 bug report

When using PostHTML Modules for templating, modifying a partial html file doesn't trigger HMR as it should.

## 🎛 Configuration (.babelrc, package.json, cli command)

My .posthtmlrc file:

```json
{
  "plugins": {
    "posthtml-modules": {
      "root": "./web"
    },
    "posthtml-expressions": {
      "root": "./web"
    }
  }
}
```

and my package.json's devDependencies:

```json
"devDependencies": {
  "@babel/core": "^7.5.4",
  "mocha": "^6.1.4",
  "parcel-bundler": "^1.12.3",
  "posthtml-expressions": "^1.1.1",
  "posthtml-modules": "^0.4.2",
  "sass": "^1.22.3"
}
```

## 🤔 Expected Behavior

Editing a partial html file included as a PostHTML module should trigger HMR.

## 😯 Current Behavior

Nothing happens when the partial html file is edited.

## 💁 Possible Solution

Maybe Parcel isn't following PostHTML modules' paths or something? I'm not sure.

## 🔦 Context

Just trying to utilize HMR when developing - it makes my life easier.

## 💻 Code Sample

https://github.com/w-biggs/1212 is the repo; https://github.com/w-biggs/1212/blob/master/web/index.html is the entry HTML file.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           |  1.12.3 |
| posthtml-modules |  0.4.2 |
| Node             | 11.14.0 |
| npm/Yarn         | 6.9.2 |
| Operating System | Ubuntu 18.04.1 LTS (WSL) |

**Comment by pugson** [comment]
Experiencing this issue as well 😥️

**Comment by sscovil** [comment]
FWIW this is still an issue, I'm experiencing it as well.

**Comment by StraightOuttaCrompton** [comment]
Why has this been marked as closed? Seems like the issue hasn't been fixed.

**Comment by BigPackie** [comment]
Or you can just update the index.html to trigger a change in the module if you use the module from index.html.

Very important: using dash/hyphen `-` inside a module as a name for a local variable binding like `{{my-variable}}` breaks the code.

**Comment by vasanthdeveloper** [comment]
yup, this issue still exists in `parcel@2.0.0-beta.1` and I can confirm that.

**Comment by schmidt-oliver** [comment]
Why has this been marked as closed? Seems like the issue hasn't been fixed.

**Comment by vsnthdev** [comment]
> Why has this been marked as closed? Seems like the issue hasn't been fixed.

The issue was closed because there wasn't any activity.

**Comment by iRON5** [comment]
The issue still [exists in 2.0.0](https://codesandbox.io/s/posthtml-modules-issue-b8ngs) (for pure reproduction download the project).

The problem is critical in this case as bundle does not update even after manual page refresh or master file save. Every change in html modules requires manual restart (no matter `watch` or `serve` command). So dev server in this case is almost redundant.

I use Parcel for small project so I use some AWFUL trick (don't repeat it at home), but if you really want to make it working [check this out](https://codesandbox.io/s/posthtml-modules-issue-awful-fix-qd799?file=/src/foo.html).

Of course it's much better to finally get the fix.
Any updates on it?

**Comment by BigPackie** [comment]
I've enjoyed using parcel but huge issues like these that does not get addressed definitely leave a bitter taste in my mouth.  I see more issues related to problems with hot reloading. I really want this project to be successful.  I've also contributed to the documentation, and It was my first contribution to a public project such as parcel. 

HMR should be prioritized more as it is a key feature of parcel for developers. It gets very frustrating to manual restart/rebuild the project from time to time because the HMR did not kick in.
